### PR TITLE
Update to RxSwift 6

### PR DESCRIPTION
**Pull request by mgray88**
This PR:
- Adds support for [RxSwift 6](https://dev.to/freak4pc/what-s-new-in-rxswift-6-2nog)
- Adds a [carthage.sh](https://github.com/Carthage/Carthage/issues/3019#issuecomment-665136323) script to workaround a Carthage issue

### CODE REVIEW

**Review by ashfurrow — APPROVED** [review]
Looks good to me 👍 CI is also erroring out on `master` so we should be okay to merge. (I'm trusting you here that Carthage will work, haha. If it doesn't at least I'll know who to ask for help!)

File: RxNimble.podspec [hunk 1]
```diff
@@ -1,6 +1,6 @@
 Pod::Spec.new do |s|
   s.name         = "RxNimble"
-  s.version      = "5.0.0"
+  s.version      = "6.0.0"
```
Comment (ashfurrow) [inline RxNimble.podspec:3]: This is actually the version number for `RxNimble`, I'll change it to 5.1.0 in our release.

Comment (mgray88) [reply RxNimble.podspec:3]: I guessed 6 since it seemed a lot of the other community repos were updating the major version number

**Comment by ashfurrow** [comment]
Okay! This has been released as 5.1.0. Thanks again!

File: RxNimble.podspec (hunk 1 above)
Comment (ashfurrow) [reply RxNimble.podspec:3]: Yeah, there's something to be said about staying in lock-step with RxSwift, but this way we can make our own breaking changes independent of them.

**Pull request merged**